app/scripts/controllers/network/network-controller.test.js

```diff
@@ -50,224 +69,4106 @@ const PRE_1559_BLOCK = {
+                  },
+                });
+
+                const promiseForInfuraIsUnblocked = waitForEvent({
+                  controller,
+                  eventName: 'infuraIsUnblocked',
+                  operation: async () => {
+                    await controller.initializeProvider();
+                  },
+                });
+
+                await expect(promiseForInfuraIsUnblocked).toNeverResolve();
+              },
+            );
+          });
+        });
+
+        it(`persists "${networkVersion}" to state as the network version of ${networkName}`, async () => {
+          await withController(
+            {
+              state: {
+                provider: {
+                  type: networkType,
+                },
+              },
+            },
+            async ({ controller, network }) => {
+              network.mockEssentialRpcCalls();
+              await withoutCallingLookupNetwork({
+                controller,
+                operation: async () => {
+                  await controller.initializeProvider();
+                },
+              });
+
+              await waitForStateChanges({
+                controller,
+                propertyPath: ['network'],
+                operation: async () => {
+                  await controller.lookupNetwork();
+                },
+              });
+
+              expect(controller.store.getState().network).toStrictEqual(
+                networkVersion,
+              );
+            },
+          );
+        });
+
+        it(`does not update the network state if it is already set to "${networkVersion}"`, async () => {
+          await withController(
+            {
+              state: {
+                provider: {
+                  type: networkType,
+                },
+              },
+            },
+            async ({ controller, network }) => {
+              network.mockEssentialRpcCalls({
+                eth_blockNumber: {
+                  times: 2,
+                },
+              });
+              await withoutCallingLookupNetwork({
+                controller,
+                operation: async () => {
+                  await controller.initializeProvider();
+                },
+              });
+
+              await waitForStateChanges({
+                controller,
+                propertyPath: ['network'],
+                operation: async () => {
+                  await controller.lookupNetwork();
+                },
+              });
+
+              const promiseForStateChanges = waitForStateChanges({
+                controller,
+                propertyPath: ['network'],
+                operation: async () => {
+                  await controller.lookupNetwork();
+                },
+              });
+
+              await expect(promiseForStateChanges).toNeverResolve();
+            },
+          );
+        });
+
+        describe('if the request for eth_getBlockByNumber responds successfully', () => {
```

Comment:
Hmm... I'm a little unclear on what case this is meant to represent. Doesn't `eth_getBlockByNumber` respond successfully in the "if the request for eth_blockNumber responds successfully" describe block tests as well?

Reply:
Oh I see, nevermind, this is testing different behavior that relates more to this call.